### ISSUE TITLE
Error: right-hand side in map store with wrong type: int (expected: address_t)

### ISSUE THREAD

**Issue by dddejan** [SRI-CSL/solidity#3]
Running on `test/libsolidity/syntaxTests/types/cyclic_dependency_check_on_consts_good.sol` results in `Error: right-hand side in map store with wrong type: int (expected: address_t)`.

**Comment by hajduakos** [comment]
Fixed in c2e0f1cad2acb0178ecd65b9ad2e90df19fbcd9a

**Comment by dddejan** [comment]
Running on the file now I get:

```
dejan$ ./solc-verify.py test/solc-verify/Issue003.sol 
Verifying A::[implicit_constructor]: OK
Verifying Issue003::[constructor]: ERROR
 - Source test/solc-verify/Issue003.sol, line 149, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 150, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 151, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 152, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 153, col 9: Assertion might not hold.
Verifying Issue003::: ERROR
 - Source test/solc-verify/Issue003.sol, line 149, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 150, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 151, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 152, col 9: Assertion might not hold.
 - Source test/solc-verify/Issue003.sol, line 153, col 9: Assertion might not hold.
Errors were found by the verifier.
```

This is unexpected.
